src/tools/yarn-plugin-tinymce.js

```diff
@@ -44,8 +44,20 @@ module.exports = {
                 ? targetName
                 : sourceName;
               const requestedFile = `${locatorPath}/node_modules/${nodeModulesPath}${sourceName}`;
-              const fileContent = crossFs.readFileSync(requestedFile);
-              crossFs.writeFileSync(pathToAssets + targetName, fileContent, 'utf8');
+
+              // exclude elements that are descendants of elements with the class .mce-preview-body only for skin.min.css
+              // without this modification, preview elements will appear with large padding and won't properly render elements past the second child of .mce-preview-body
+              let fileContent = crossFs.readFileSync(requestedFile);
+              if (Buffer.isBuffer(fileContent)) {
+                fileContent = fileContent.toString('utf8');
+              }
+              if (sourceName === 'skin.min.css') {
```

Comment:
doing a regex on the css file from npm seems like a wrong approach to me. Isn't it possible to override this css in our own css?

Reply:
Yes, it would be. I'm going to look over which rules need to be overwritten manually by modifying `main.scss` with `!important` CSS rules.

Reply:
I can't use the array approach because the `div` element doesn't have an `iframe`. As a demonstration, I added the `content_style` setting to 'h1 {color:red;}'. While the h1 element in the editor is styled red, the h1 element in the preview is not.
![image](https://github.com/user-attachments/assets/53aac7d0-cb78-4a34-9555-74ac73821c9b)
![image](https://github.com/user-attachments/assets/723c1507-6e7b-4a67-bfd4-43a774ab74ea)

Similarly, `tinymce_content.min.css` doesn't seem to apply to the div element, so I would need to instead edit `elabftw.min.css` (through `main.scss`) and put in the styling rules with `!important`.

Alternatively, I can try to modify the CSS file in `yarn-plugin-tinymce.js` that doesn't rely on using regex rules so that any new rules that may be added to the skin file (if it changes in TinyMCE 7, for example) can be automatically changed to not apply in `.tinymce-preview-body`.

Reply:
Having a few rules with `!important` in the main css file (or better a _tinymce_preview.scss partial file imported in the main) seems reasonable. I'd prefer not to mess with the yarn plugin if possible. Having css in a scss file that clearly indicates why it's here and why the rules need to be !important sounds good to me.

Reply:
In the pursuit of finding a solution that doesn't have to be updated every time a skin stylesheet changes, I manipulated the `CSSStyleSheet` DOM object to basically append `:not(tinymce-preview_body` after TinyMCE initialization to each rule in `skin.min.css` without actually changing the underlying CSS file (see commit 65342b804c50f3ffcdadbd9b4aee975ea2e8a529). Let me know if this approach is adequate.